openmc/data/multipole.py

```diff
@@ -129,6 +143,608 @@ def _broaden_wmp_polynomials(E, dopp, n):
+                                    energy[0], energy[-1], mt))
+            plt.savefig(fig_file)
+            plt.close()
+            if log:
+                print("Saved figure: {}".format(fig_file))
+
+    return (mp_poles, mp_residues)
+
+
+def vectfit_nuclide(endf_file, njoy_error=5e-4, vf_pieces=None,
+                    log=False, path_out=None, mp_filename=None, **kwargs):
+    r"""Generate multipole data for a nuclide from ENDF.
+
+    Parameters
+    ----------
+    endf_file : str
+        Path to ENDF evaluation
+    njoy_error : float, optional
+        Fractional error tolerance for processing point-wise data with NJOY
+    vf_pieces : integer, optional
+        Number of equal-in-momentum spaced energy pieces for data fitting
+    log : bool or int, optional
+        Whether to print running logs (use int for verbosity control)
+    path_out : str, optional
+        Path to write out mutipole data file and vector fitting figures
+    mp_filename : str, optional
+        File name to write out multipole data
+    **kwargs
+        Keyword arguments passed to :func:`openmc.data.multipole._vectfit_xs`
+
+    Returns
+    -------
+    mp_data
+        Dictionary containing necessary multipole data of the nuclide
+
+    """
+
+    # ======================================================================
+    # PREPARE POINT-WISE XS
+
+    # make 0K ACE data using njoy
+    if log:
+        print("Running NJOY to get 0K point-wise data (error={})...".format(njoy_error))
+
+    nuc_ce = IncidentNeutron.from_njoy(endf_file, temperatures=[0.0],
+             error=njoy_error, broadr=False, heatr=False, purr=False)
+
+    if log:
+        print("Parsing cross sections within resolved resonance range...")
+
+    # Determine upper energy: the lower of RRR upper bound and first threshold
+    endf_res = IncidentNeutron.from_endf(endf_file).resonances
+    if hasattr(endf_res, 'resolved') and \
+       hasattr(endf_res.resolved, 'energy_max') and \
+       type(endf_res.resolved) is not ResonanceRange:
+        E_max = endf_res.resolved.energy_max
+    elif hasattr(endf_res, 'unresolved') and \
+         hasattr(endf_res.unresolved, 'energy_min'):
+        E_max = endf_res.unresolved.energy_min
+    else:
+        E_max = nuc_ce.energy['0K'][-1]
+    E_max_idx = np.searchsorted(nuc_ce.energy['0K'], E_max, side='right') - 1
+    for mt in nuc_ce.reactions:
+        if hasattr(nuc_ce.reactions[mt].xs['0K'], '_threshold_idx'):
+            threshold_idx = nuc_ce.reactions[mt].xs['0K']._threshold_idx
+            if 0 < threshold_idx < E_max_idx:
+                E_max_idx = threshold_idx
+
+    # parse energy and cross sections
+    energy = nuc_ce.energy['0K'][:E_max_idx + 1]
+    E_min, E_max = energy[0], energy[-1]
+    n_points = energy.size
+    total_xs = nuc_ce[1].xs['0K'](energy)
+    try:
+        elastic_xs = nuc_ce[2].xs['0K'](energy)
+    except KeyError:
+        elastic_xs = np.zeros_like(total_xs)
```

Comment:
Every nuclide should have elastic scattering, so if it doesn't appear something is seriously wrong
```suggestion
    elastic_xs = nuc_ce[2].xs['0K'](energy)
```

Reply:
Agree. I've never seen a nuclide without `elastic xs`. Updated.